### ISSUE TITLE
Multi IP / IPv6 support

### ISSUE THREAD

**Issue by igalic** [ployground/ploy_ezjail#10]
from the [`ezjail-admin(8)`](https://www.freebsd.org/cgi/man.cgi?ezjail-admin) page:

>     ipaddress[,ipaddress2,...]
>             The IP address or addresses of the jail. Since FreeBSD 7.2, it is
>             possible to assign several several IPv4 or IPv6 addresses to a
>             jail, by separating them with commas. Previous versions of
>             FreeBSD allowed only a single IPv4 address per jail.

however, when trying to start this jail:

```dosini
[ez-instance:test]
master = jailhost
ip = 10.0.0.3,2a03:b0c0:3:d0::3a4d:c002
```

we get the following stack-trace:

```
% ploy start test
INFO: Creating instance 'test'
Traceback (most recent call last):
  File "/home/igalic/src/me/meenix/bsdploy/bin/ploy", line 11, in <module>
    sys.exit(ploy())
  File "/home/igalic/src/me/meenix/bsdploy/local/lib/python2.7/site-packages/ploy/__init__.py", line 561, in ploy
    return ctrl(argv)
  File "/home/igalic/src/me/meenix/bsdploy/local/lib/python2.7/site-packages/ploy/__init__.py", line 553, in __call__
    args.func(sub_argv, args.func.__doc__)
  File "/home/igalic/src/me/meenix/bsdploy/local/lib/python2.7/site-packages/ploy/__init__.py", line 288, in cmd_start
    result = instance.start(overrides)
  File "/home/igalic/src/me/meenix/bsdploy/local/lib/python2.7/site-packages/ploy_ezjail/__init__.py", line 150, in start
    jails = self.master.ezjail_admin('list')
  File "/home/igalic/src/me/meenix/bsdploy/local/lib/python2.7/site-packages/ploy_ezjail/__init__.py", line 489, in ezjail_admin
    jails[entry.pop('name')] = entry
KeyError: 'name'
```

Creating the jail is possible, however, further processing then is hampered by our `ezjail-admin list` parsing code:
```python
        elif command == 'list':
            rc, out, err = self._ezjail_admin('list')
            if rc:
                msg = out.strip() + '\n' + err.strip()
                raise EzjailError(msg.strip())
            lines = out.splitlines()
            if len(lines) < 2:
                raise EzjailError("ezjail-admin list output too short:\n%s" % out.strip())
            headers = self.ezjail_admin_list_headers
            jails = {}
            for line in lines[2:]:
                line = line.strip()
                if not line:
                    continue
                entry = dict(zip(headers, line.split()))
                jails[entry.pop('name')] = entry
            return jails
```
which cannot deal with output like this:
```
ezjail-admin list 
STA JID  IP              Hostname                       Root Directory
--- ---- --------------- ------------------------------ ------------------------
ZS  N/A  10.0.0.3        test                           /usr/jails/test
    N/A  2a03:b0c0:3:d0::3a4d:c002
```

**Comment by igalic** [comment]
in addition to being hard to parse, this configuration is also wrong.

the correct configuration should be:

```dosini
[ez-instance:test]
master = jailhost
ip = lo1|10.0.0.3,vtnet0|2a03:b0c0:3:d0::3a4d:c002
```

**Comment by igalic** [comment]
I have a very naïve patch for this:
```diff
diff --git a/ploy_ezjail/__init__.py b/ploy_ezjail/__init__.py
index 86a179c..f542168 100644
--- a/ploy_ezjail/__init__.py
+++ b/ploy_ezjail/__init__.py
@@ -485,12 +485,17 @@ class Master(BaseMaster):
                 raise EzjailError("ezjail-admin list output too short:\n%s" % out.strip())
             headers = self.ezjail_admin_list_headers
             jails = {}
+            prev_entry = None
             for line in lines[2:]:
                 line = line.strip()
                 if not line:
                     continue
-                entry = dict(zip(headers, line.split()))
-                jails[entry.pop('name')] = entry
+                if line.startswith('\s+'):
+                    jails[prev_entry][ip] = [jails[prev_entry][ip], line.split()[1]]
+                else:
+                    entry = dict(zip(headers, line.split()))
+                    prev_entry = entry.pop('name')
+                    jails[prev_entry] = entry
             return jails
         elif command == 'start':
             rc, out, err = self._ezjail_admin(
```

**Comment by igalic** [comment]
this now allows us to create the jail, however, we cannot ssh into it,
```
ploy ssh patch                                                                           (48s 618ms)  
ERROR: Exception: ('nohup ssh -o StrictHostKeyChecking=yes -o UserKnownHostsFile=/home/igalic/src/me/meenix/etc/known_hosts -l root -p 22 meenirx -W lo1|10.0.0.4,vtnet0|2a03:b0c0:3:d0::3a4d:c003:22', 'Broken pipe')
ERROR: Traceback (most recent call last):
ERROR:   File "/home/igalic/src/me/meenix/bsdploy/local/lib/python2.7/site-packages/paramiko/transport.py", line 1747, in run
ERROR:     self.packetizer.write_all(b(self.local_version + '\r\n'))
ERROR:   File "/home/igalic/src/me/meenix/bsdploy/local/lib/python2.7/site-packages/paramiko/packet.py", line 288, in write_all
ERROR:     n = self.__socket.send(out)
ERROR:   File "/home/igalic/src/me/meenix/bsdploy/local/lib/python2.7/site-packages/paramiko/proxy.py", line 73, in send
ERROR:     raise ProxyCommandFailure(' '.join(self.cmd), e.strerror)
ERROR: ProxyCommandFailure: ('nohup ssh -o StrictHostKeyChecking=yes -o UserKnownHostsFile=/home/igalic/src/me/meenix/etc/known_hosts -l root -p 22 meenirx -W lo1|10.0.0.4,vtnet0|2a03:b0c0:3:d0::3a4d:c003:22', 'Broken pipe')
ERROR: 
ERROR: Failed to connect to ez-instance:patch (lo1|10.0.0.4,vtnet0|2a03:b0c0:3:d0::3a4d:c003)
ERROR: username: 'root'
ERROR: port: 22
ERROR: sock: <paramiko.proxy.ProxyCommand object at 0x7f0c45d12ad0>
ERROR: Couldn't validate fingerprint for ssh connection.
ERROR: ('nohup ssh -o StrictHostKeyChecking=yes -o UserKnownHostsFile=/home/igalic/src/me/meenix/etc/known_hosts -l root -p 22 meenirx -W lo1|10.0.0.4,vtnet0|2a03:b0c0:3:d0::3a4d:c003:22', 'Broken pipe')
ERROR: Is the instance finished starting up?
```

or configure it:

```
(bsdploy)  bsdploy  % ~/s/m/meenix      ploy configure patch                                                                (1m 5s 628ms)  
INFO: Using playbook at '/home/igalic/src/me/meenix/jailhost-patch.yml'.

PLAY [jailhost-patch] ********************************************************* 

GATHERING FACTS *************************************************************** 
ERROR: Exception: ('nohup ssh -o StrictHostKeyChecking=yes -o UserKnownHostsFile=/home/igalic/src/me/meenix/etc/known_hosts -l root -p 22 meenirx -W lo1|10.0.0.4,vtnet0|2a03:b0c0:3:d0::3a4d:c003:22', 'Broken pipe')
ERROR: Traceback (most recent call last):
ERROR:   File "/home/igalic/src/me/meenix/bsdploy/local/lib/python2.7/site-packages/paramiko/transport.py", line 1747, in run
ERROR:     self.packetizer.write_all(b(self.local_version + '\r\n'))
ERROR:   File "/home/igalic/src/me/meenix/bsdploy/local/lib/python2.7/site-packages/paramiko/packet.py", line 288, in write_all
ERROR:     n = self.__socket.send(out)
ERROR:   File "/home/igalic/src/me/meenix/bsdploy/local/lib/python2.7/site-packages/paramiko/proxy.py", line 73, in send
ERROR:     raise ProxyCommandFailure(' '.join(self.cmd), e.strerror)
ERROR: ProxyCommandFailure: ('nohup ssh -o StrictHostKeyChecking=yes -o UserKnownHostsFile=/home/igalic/src/me/meenix/etc/known_hosts -l root -p 22 meenirx -W lo1|10.0.0.4,vtnet0|2a03:b0c0:3:d0::3a4d:c003:22', 'Broken pipe')
ERROR: 
ERROR: Failed to connect to ez-instance:patch (lo1|10.0.0.4,vtnet0|2a03:b0c0:3:d0::3a4d:c003)
ERROR: username: 'root'
ERROR: port: 22
ERROR: sock: <paramiko.proxy.ProxyCommand object at 0x7f8270064fd0>
fatal: [jailhost-patch] => Couldn't validate fingerprint for 'ez-instance:patch': ('nohup ssh -o StrictHostKeyChecking=yes -o UserKnownHostsFile=/home/igalic/src/me/meenix/etc/known_hosts -l root -p 22 meenirx -W lo1|10.0.0.4,vtnet0|2a03:b0c0:3:d0::3a4d:c003:22', 'Broken pipe')

TASK: [configure does't work] ************************************************* 
FATAL: no hosts matched or all hosts have already failed -- aborting
```

**Comment by fschulze** [comment]
the [get_host](https://github.com/ployground/ploy_ezjail/blob/master/ploy_ezjail/__init__.py#L63) method needs to be changed for that. Ploy itself isn't aware of multiple IP addresses yet. So you would have to parse the ``ip = ...`` line in that method and only return the IP that will be used for ssh. I suggest to use the first one.

**Comment by igalic** [comment]
This is something we can fix by fixing `get_host()`

**Comment by igalic** [comment]
@fschulze i have, indeed, a patch already

**Comment by fschulze** [comment]
Cool! I left some comments on the PRs.

**Comment by igalic** [comment]
i think we can close this now, esp now with tests added
